File: hotel_api/hotel_data/dto/RoomTypeDto.cs
namespace hotel_data.dto;

public class RoomTypeDto
{
    public RoomTypeDto(Guid? roomTypeId, string roomTypeName, Guid createdBy, DateTime createdAt)
    {
        roomTypeID = roomTypeId;
        this.roomTypeName = roomTypeName;
        this.createdBy = createdBy;
        this.createdAt = createdAt;
    }

    public Guid? roomTypeID { get; set; }
    public string roomTypeName { get; set; }
    public Guid createdBy { get; set; }
    public DateTime createdAt { get; set; }
    
}

File: hotel_mobile/app/src/main/java/com/example/hotel_mobile/Dto/RoomTypeDto.kt
package com.example.hotel_mobile.Dto

import com.example.hotel_mobile.services.kSerializeChanger.LocalDateTimeKserialize
import com.example.hotel_mobile.services.kSerializeChanger.UUIDKserialize
import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class RoomTypeDto(
    @Serializable(with = UUIDKserialize::class)
    val roomTypeID: UUID?,
    val roomTypeName: String,
    val imagePath: String? = null,
    val createdBy: String,
    @Serializable(with = LocalDateTimeKserialize::class)
    val createdAt: LocalDateTime,
    val isDeleted: Boolean
)
